app/controllers/api/v2/play_roles_controller.rb

```diff
@@ -0,0 +1,93 @@
+module Api
+  module V2
+    # API controller for Ansible Plays
+    class PlayRolesController < ::Api::V2::BaseController
+      include ::Api::Version2
+      include ForemanTasks::Triggers
+
+      rescue_from ActiveRecord::RecordNotFound, :with => :resource_not_found
+
+      before_action :find_hosts, :only => [:play_roles_on_host]
```

Comment:
@Gerrit91 You got it right, exactly that - adding the methods to the hosts/hostgroups controllers allow you to use `find_resource` and also the API paths will be similar to those of Puppet/Salt and other paths in Foreman.